paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/lookup/FileStoreLookupFunction.java

```diff
@@ -145,9 +154,41 @@ private void open() throws Exception {
                         options.get(LOOKUP_CACHE_ROWS));
         this.nextLoadTime = -1;
         this.streamingReader = new TableStreamingReader(table, projection, this.predicate);
+        bulkLoad(options);
+    }
 
-        // do first load
-        refresh();
+    private void bulkLoad(Options options) throws Exception {
+        BinaryExternalSortBuffer bulkLoadSorter =
```

Comment:
Is this sort necessary to the primary key index lookup? Since, it's already sorted